Sources/Kitura/CodableRouter.swift

```diff
@@ -417,6 +479,48 @@ extension Router {
         }
     }
 
+    // Get w/Optional Query Parameters
+    fileprivate func getSafely<Q: QueryParams, O: Codable>(_ route: String, handler: @escaping (Q?, @escaping CodableArrayResultClosure<O>) -> Void) {
+        registerGetRoute(route: route, id: false, outputtype: O.self)
+        get(route) { request, response, next in
+            Log.verbose("Received GET (plural) type-safe request with Query Parameters")
+            Log.verbose("Query Parameters: \(request.queryParameters)")
+            do {
+                var query: Q? = nil
+                let queryParameters = request.queryParameters
+                if queryParameters.count > 0 {
+                    query = try QueryDecoder(dictionary: queryParameters).decode(Q.self)
+                }
+                handler(query, CodableHelpers.constructOutResultHandler(response: response, completion: next))
```

Comment:
Would it be worth extracting the building of the query object into a helper?  Something like
```swift
func decodeQueryParams<Q: QueryParams>(_ request: RouterRequest) throws -> Q?
```
That way if we need more permutations of route patterns that accept `QueryParams?` there is less code to duplicate.

Ideally we could go a step further by avoiding the need for the do/catch, but that might take some more thinking (perhaps the helper could take a closure to invoke on success, or set the response status on failure).